satpy/readers/viirs_sdr.py

```diff
@@ -365,7 +366,6 @@ def get_dataset(self, dataset_id, ds_info):
         file_units = self.get_file_units(dataset_id, ds_info)
         output_units = ds_info.get("units", file_units)
         factors = self.adjust_scaling_factors(factors, file_units, output_units)
-
         if factors is not None:
             data = self.scale_swath_data(data, factors)
```

Comment:
@mraspaud I set a breakpoint in my debugger on this if statement when factors *is* None and it never hits when running the viirs_sdr tests. This if is only skipped in the case that the YAML doesn't have any units specified and factors is None. Or the YAML has file_units and units specified as the same thing and factors is still None. This is currently not configured this way so I'd have to make something up to test it. Bottom line: I found issues in the YAML because I had removed this if statement, it is now no longer needed, but it is needed to safe guard against bad configuration.

Reply:
Interesting. So could we add a warning here to tell the user there is something fishy in the yaml then?

Reply:
I thought about it last night and I think we should keep this as is. It is completely valid for someone to specify the units, for the units to match what is in the file, and there to be no factors in the file.

Reply:
Sounds good.